kyo-core/shared/src/main/scala/kyo/scheduler/IOTask.scala

```diff
@@ -49,7 +49,7 @@ private[kyo] class IOTask[T](
                         k.command match
                             case Promise(p) =>
                                 this.interrupts(p)
-                                val runtime = (clock.currentMillis() - startMillis).toInt
+                                val runtime = (clock.currentMillis() - startMillis + this.runtime()).toInt
```

Comment:
This scenario happens when the fiber/io task suspends because of an async boundary. Here, `runtime` should contain the time spent on the current execution plus all of the scheduling time the fiber got before it.